File: 백준/Silver/2606. 바이러스/바이러스.py
num = int(input())

ssang = int(input())


graph = [[] * (num+1) for i in range(num+1)]

for i in range(ssang):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

visited = [0 for i in range(num+1)]

def dfs(n):
    if visited[n]:
        return

    visited[n] = True
    global cnt
    cnt += 1

    for neighbor in graph[n]:
        dfs(neighbor)

cnt = 0
dfs(1)
print(cnt-1)

File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <iostream>
#include <vector>
using namespace std;

bool visited[101];
int cnt = 0;
vector<int> arr[101];

void dfs(int now) {
	if (visited[now])return;
	visited[now] = true;
	cnt++;
	for (int i = 0; i < arr[now].size(); i++) {
		int nxt = arr[now][i];
		dfs(nxt);
	}
}

int main() {
	int n;
	int m;
	cin >> n >> m;
	for (int i = 0; i < m; i++) {
		int u, v;
		cin >> u >> v;
		arr[u].push_back(v);
		arr[v].push_back(u);
	}
	dfs(1);
	cout << cnt-1 << "\n";
}